scripts/alert_emails/send_emails.ts

```diff
@@ -162,73 +168,95 @@ async function setLastSnapshotNumber(
-                );
-                await delay(rateLimitReset * 1000);
-              }
-              // Comment out this code if you are developing locally
-              await db
-                .collection(
-                  `snapshots/${currentSnapshot}/locations/${fips}/emails/`,
-                )
-                .doc(doc.id)
-                .set({
-                  sentAt: admin.firestore.FieldValue.serverTimestamp(),
-                });
-            })
-            .catch(async (err: CampaignMonitorError) => {
-              // if the email is invalid, move the document to the invalid collection
-              if (
-                err.Code === CM_INVALID_EMAIL_ERROR_CODE &&
-                err.Message === CM_INVALID_EMAIL_MESSAGE
-              ) {
-                invalidEmailCount += 1;
-                const currentData = (
-                  await db.collection('alerts-subscriptions').doc(doc.id).get()
-                ).data();
-                if (currentData) {
-                  await db
-                    .collection('invalid-alerts-subscriptions')
-                    .doc(doc.id)
-                    .set(currentData);
-                  await db
-                    .collection('alerts-subscriptions')
-                    .doc(doc.id)
-                    .delete();
-                }
-              } else {
-                errorCount += 1;
-              }
-            });
-        }
-      });
+      .get();
+    return querySnapshot.docs.map(emailDoc => emailDoc.id);
+  };
+
+  async function onInvalidEmail(email: string) {
+    invalidEmailCount += 1;
+    const querySnapshot = await db
+      .collection('alerts-subscriptions')
+      .doc(email)
+      .get();
+    const currentData = querySnapshot.data();
+    if (currentData) {
+      await db
+        .collection('invalid-alerts-subscriptions')
+        .doc(email)
+        .set(currentData);
+      await db.collection('alerts-subscriptions').doc(email).delete();
+    }
+  }
+
+  async function onEmailSent(email: string, fips: string) {
+    emailSent += 1;
+    uniqueEmailAddress[email] = (uniqueEmailAddress[email] || 0) + 1;
+    locationsWithEmails[fips] = (locationsWithEmails[fips] || 0) + 1;
+    // comment out this code if you are developing locally
+    return db
+      .collection(`snapshots/${currentSnapshot}/locations/${fips}/emails/`)
+      .doc(email)
+      .set({ sentAt: admin.firestore.FieldValue.serverTimestamp() });
+  }
+
+  async function sendAlertEmail(email: string, fips: string) {
+    const locationAlert = locationsWithAlerts[fips];
+    const sendData = generateSendData(email, locationAlert);
+
+    try {
+      const response = await sendEmail(api, sendData, dryRun);
+      await onEmailSent(email, fips);
+      return response;
+    } catch (err) {
+      if (isInvalidEmail(err)) {
+        await onInvalidEmail(email);
+      } else {
+        errorCount += 1;
```

Comment:
I guess this is all the old code did, but maybe we should `console.error(err)` here as well?